contracts/Exchange.sol

```diff
@@ -160,48 +161,31 @@ contract Exchange is
             IAccountBalance(_accountBalance).getTakerPositionSize(params.trader, params.baseToken);
 
         bool isPartialClose;
-        bool isOverPriceLimit = _isOverPriceLimit(params.baseToken);
-        // if over price limit when
-        // 1. closing a position, then partially close the position
-        // 2. else then revert
+
         if (params.isClose && takerPositionSize != 0) {
             // if trader is on long side, baseToQuote: true, exactInput: true
             // if trader is on short side, baseToQuote: false (quoteToBase), exactInput: false (exactOutput)
-            // simulate the tx to see if it _isOverPriceLimit; if true, can partially close the position only once
-            // if this is not the first tx in this timestamp and it's already over limit,
-            // then use _isOverPriceLimit is enough
+            // simulate the tx to see if it's over the price limit; if true, can partially close the position
             if (
-                isOverPriceLimit ||
                 _isOverPriceLimitBySimulatingClosingPosition(
                     params.baseToken,
                     takerPositionSize < 0, // it's a short position
                     params.amount // it's the same as takerPositionSize but in uint256
                 )
             ) {
-                uint256 timestamp = _blockTimestamp();
-                // EX_AOPLO: already over price limit once
-                require(timestamp != _lastOverPriceLimitTimestampMap[params.trader][params.baseToken], "EX_AOPLO");
-
-                _lastOverPriceLimitTimestampMap[params.trader][params.baseToken] = timestamp;
-
                 uint24 partialCloseRatio = IClearingHouseConfig(_clearingHouseConfig).getPartialCloseRatio();
                 params.amount = params.amount.mulRatio(partialCloseRatio);
                 isPartialClose = true;
             }
-        } else {
```

Comment:
Did we forget to remove this before? 
I guess the reason is since we require `EX_OPLAS`, we can always make sure the price will never be over the limit before swap

Reply:
Yes, we forgot to remove this in v2.4.1.
As you said, after the hotfix of release v2.4.1, each trade will not exceed the price limit whatever it's opening, reducing or closing a position. Thus, it's a redundant step to check the price limit before swap.